omod-1.10/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_10/OrderResource1_10.java

```diff
@@ -243,12 +248,52 @@ protected PageableResult doSearch(RequestContext context) {
 				filterByType(orders, context.getType());
 			}
 			
+			if (StringUtils.isNotBlank(sortParam)) {
+				sortOrdersBasedOnDateActivatedOrDateStopped(orders, sortParam, status);
+			}
+			
 			return new NeedsPaging<Order>(orders, context);
 		}
 		
 		return new EmptySearchResult();
 	}
 	
+	private static void sortOrdersBasedOnDateActivatedOrDateStopped(List<Order> orders, String sortOrder, final String status) {
+		//sort in ascending order using dateactivated or datestopped depending on the status
+		if (sortOrder.compareToIgnoreCase("ASC") == 0) {
+			
+			Collections.sort(orders, new Comparator<Order>() {
+				
+				@Override
+				public int compare(Order o1, Order o2) {
+					if (status.compareToIgnoreCase("inactive") == 0) {
+						return o1.getDateStopped().compareTo(o2.getDateStopped());
+					}
+					else {
+						return o1.getDateActivated().compareTo(o2.getDateActivated());
+					}
+				}
+			});
+		}
+		
+		//sort in descending order using dateactivated or datestopped depending on the status
+		else {
+			Collections.sort(orders, new Comparator<Order>() {
+				
+				@Override
+				public int compare(Order o1, Order o2) {
+					if (status.compareToIgnoreCase("inactive") == 0) {
+						return o2.getDateStopped().compareTo(o1.getDateStopped());
+					}
+					else {
+						return o2.getDateActivated().compareTo(o1.getDateActivated());
+					}
+					
+				}
+			});
+		}
+	}
+	
```

Comment:
I would suggest simplifying lines 263 - 295 further
```
Collections.sort(orders, new Comparator<Order>() {		
				@Override
				public int compare(Order o1, Order o2) {
					if (status.compareToIgnoreCase("inactive") == 0) {
						return sortOrder.compareToIgnoreCase("ASC") == 0 ? o1.getDateStopped().compareTo(o2.getDateStopped()) : o2.getDateStopped().compareTo(o1.getDateStopped());
					}
					else {
						return sortOrder.compareToIgnoreCase("ASC") == 0 ? o1.getDateActivated().compareTo(o2.getDateActivated()) : o2.getDateActivated().compareTo(o1.getDateActivated());
					}
				}
			});
```
I am just worried about code duplication. I may be wrong though

Reply:
I'd agree with this change a well.  Thanks @larrystone !